src/messages/MOSDOp.h

```diff
@@ -242,14 +236,26 @@ struct ceph_osd_request_head {
       ::encode_nohead(snaps, payload);
     } else {
       header.version = HEAD_VERSION;
-      ::encode(client_inc, payload);
-      ::encode(osdmap_epoch, payload);
-      ::encode(flags, payload);
-      ::encode(mtime, payload);
-      ::encode(reassert_version, payload);
 
-      ::encode(oloc, payload);
-      ::encode(pgid, payload);
+      // new, reordered message encoding
+      if (features & CEPH_FEATURE_NEW_ENCODING) {
+        ::encode(pgid, payload);
+        ::encode(osdmap_epoch, payload);
+        ::encode(flags, payload);
+        ::encode(client_inc, payload);
+        ::encode(mtime, payload);
+        ::encode(reassert_version, payload);
+        ::encode(oloc, payload);
+      } else {
+        ::encode(client_inc, payload);
+        ::encode(osdmap_epoch, payload);
+        ::encode(flags, payload);
+        ::encode(mtime, payload);
+        ::encode(reassert_version, payload);
+        ::encode(oloc, payload);
+        ::encode(pgid, payload);
```

Comment:
So we're here encoding in the classic order because the other end doesn't understand the new order, but as part of that we also should set the message version to be the older one (5, I think?). You just need to overwrite head.version with the appropriate value.